### PR TITLE
New: Kit-Library - Tests [ED-3135]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

**Review by matipojo — APPROVED** [review]

File: tests/phpunit/traits/rest-trait.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
```
Comment (iNewLegend) [inline tests/phpunit/traits/rest-trait.php:1]: Great :)

File: tests/phpunit/elementor/core/app/kit-library/data/kits/test-controller.php [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+		$this->act_as_admin();
+		$this->app_mock->method( 'get_all' )->willReturn( $this->get_kits_api_mock() );
+		$this->app_mock->method( 'get_manifest' )->willReturn( $this->get_manifest_api_mock() );
+
+		$this->data_manager->register_controller( Controller::class );
+
+		// Act
+		$result = $this->http_get( "kits/id_1", [ 'force' => true ] );
+
+//		var_dump($result);die;
```
Comment (iNewLegend) [inline tests/phpunit/elementor/core/app/kit-library/data/kits/test-controller.php:96]: var_dump

**Review by iNewLegend — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/app/kit-library/data/kits/test-controller.php (hunk 2 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/app/kit-library/data/kits/test-controller.php:96]: ops... 😸

**Review by iNewLegend — APPROVED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+			        'code' => 200,
+			    ],
+			    'body' => 'null',
+			] );
+
+		// Act
+		$result = $this->app_stub->proxy_http_request( 'POST', 'test' );
+
+		// Assert
+		$this->assertEquals( 1, $result );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:98]: Who returns the `1`?

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+			    $this->equalTo( Mock_App::BASE_URL . '/' . $endpoint ),
+			    $this->equalTo( [
+			        'headers' => [
+			        	'access-token' => 'access_token_test',
+				        'app' => 'mock-app',
+				        'client-id' => 'client_id_test',
+				        'endpoint' => $endpoint,
+				        'home-url' => 'http://example.org/',
+				        'local-id' => $user->ID,
+						'site-key' => 'site_key_test',
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:172]:
```suggestion
					'site-key' => 'site_key_test',
```

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+			'headers' => [
+				'a' => '1',
+			],
+			'body' => [
+				'b' => '2',
+			]
+		] );
+
+		// Assert
+		$this->assertEquals( (object) [ 'status' => 'success' ], $result );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:202]: Where is the body (b=2)?

Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:202]: I didn't understand the test (why or how will the connect headers affect it?)

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 3 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:98]: When the body is a string with the value of 'null' and the code change the value to `true`, then after the json_decode it becomes `1`. this is the behavior that was before this change.

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 5 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:202]: There is a `with` method that checks before that the b=2 was sent to the API, the status success is the return response from the API.
I will add some comments

Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:202]: The `with` method of the mock checks that the right parameters are sent to the API. 
I will add some comments

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 4 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:174]: Done

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 5 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:204]: Done

Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:204]: Done

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 3 above)
Comment (matipojo) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:99]: Explain in code.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request merged**